Upstream build failing for 2 weeks

https://github.com/pystatgen/sgkit/actions?query=workflow%3A%22Upstream%22+branch%3Amaster

The failure is due to https://github.com/pydata/xarray/pull/4940

```
____________________ test_hwep_dataset__precomputed_counts _____________________

ds_neq = <xarray.Dataset>
Dimensions:             (alleles: 2, ploidy: 2, samples: 1000, variants: 50)
Dimensions without coord..... 0 0 0 0
    call_genotype_mask  (variants, samples, ploidy) bool False False ... False
Attributes:
    contigs:  [0]

    def test_hwep_dataset__precomputed_counts(ds_neq: Dataset) -> None:
        ds = ds_neq
        ac = ds["call_genotype"].sum(dim="ploidy")
        cts = [1, 0, 2]  # arg order: hets, hom1, hom2
        gtc = xr.concat([(ac == ct).sum(dim="samples") for ct in cts], dim="counts").T
        ds = ds.assign(**{"variant_genotype_counts": gtc})
        p = hwep_test(ds, genotype_counts="variant_genotype_counts", merge=False)
>       assert np.all(p < 1e-8)

sgkit/tests/test_hwe.py:143: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
<__array_function__ internals>:5: in all
    ???
/opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/numpy/core/fromnumeric.py:2438: in all
    return _wrapreduction(a, np.logical_and, 'all', axis, None, out,
/opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/numpy/core/fromnumeric.py:85: in _wrapreduction
    return reduction(axis=axis, out=out, **passkwargs)
/opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/xarray/core/common.py:101: in wrapped_func
    return self.reduce(func, dim, numeric_only=numeric_only, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <xarray.Dataset>
Dimensions:              (variants: 50)
Dimensions without coordinates: variants
Data variables:
    variant_hwe_p_value  (variants) bool dask.array<chunksize=(50,), meta=np.ndarray>
func = <function _dask_or_eager_func.<locals>.f at 0x7f825fcef550>, dim = None
keep_attrs = None, keepdims = False, numeric_only = False
kwargs = {'axis': None, 'out': None}

    def reduce(
        self,
        func: Callable,
        dim: Union[Hashable, Iterable[Hashable]] = None,
        keep_attrs: bool = None,
        keepdims: bool = False,
        numeric_only: bool = False,
        **kwargs: Any,
    ) -> "Dataset":
        """Reduce this dataset by applying `func` along some dimension(s).
    
        Parameters
        ----------
        func : callable
            Function which can be called in the form
            `f(x, axis=axis, **kwargs)` to return the result of reducing an
            np.ndarray over an integer valued axis.
        dim : str or sequence of str, optional
            Dimension(s) over which to apply `func`.  By default `func` is
            applied over all dimensions.
        keep_attrs : bool, optional
            If True, the dataset's attributes (`attrs`) will be copied from
            the original object to the new one.  If False (default), the new
            object will be returned without attributes.
        keepdims : bool, default: False
            If True, the dimensions which are reduced are left in the result
            as dimensions of size one. Coordinates that use these dimensions
            are removed.
        numeric_only : bool, optional
            If True, only apply ``func`` to variables with a numeric dtype.
        **kwargs : Any
            Additional keyword arguments passed on to ``func``.
    
        Returns
        -------
        reduced : Dataset
            Dataset with this object's DataArrays replaced with new DataArrays
            of summarized data and the indicated dimension(s) removed.
        """
        if "axis" in kwargs:
>           raise ValueError(
                "passing 'axis' to Dataset reduce methods is ambiguous."
                " Please use 'dim' instead."
            )
E           ValueError: passing 'axis' to Dataset reduce methods is ambiguous. Please use 'dim' instead.

/opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/xarray/core/dataset.py:4693: ValueError
```

--- a/sgkit/tests/test_hwe.py
+++ b/sgkit/tests/test_hwe.py
@@ -139,7 +139,9 @@ def test_hwep_dataset__precomputed_counts(ds_neq: Dataset) -> None:
     cts = [1, 0, 2]  # arg order: hets, hom1, hom2
     gtc = xr.concat([(ac == ct).sum(dim="samples") for ct in cts], dim="counts").T
     ds = ds.assign(**{"variant_genotype_counts": gtc})
-    p = hwep_test(ds, genotype_counts="variant_genotype_counts", merge=False)
+    p = hwep_test(ds, genotype_counts="variant_genotype_counts", merge=False)[
+        "variant_hwe_p_value"
+    ].values
     assert np.all(p < 1e-8)
 
 

